lib/recurly/client.rb

```diff
@@ -42,17 +46,18 @@ class Client
     # @param subdomain [String] Optional subdomain for the site you wish to be scoped to. Providing this makes all the `site_id` parameters optional.
     def initialize(api_key:, site_id: nil, subdomain: nil, **options)
       set_site_id(site_id, subdomain)
+      set_api_key(api_key)
       set_options(options)
-      set_faraday_connection(api_key)
 
       # execute block with this client if given
       yield(self) if block_given?
     end
 
     def next_page(pager)
-      req = HTTP::Request.new(:get, pager.next, nil)
-      faraday_resp = run_request(req, headers)
-      handle_response! req, faraday_resp
+      path = URI(pager.next).path
```

Comment:
Is there a reason we need to parse into URI first? This line drops the query params:

```ruby
irb(main):003:0> URI("/accounts?cursor=xyz&limit=20&sort=created_at").path
=> "/accounts"
```

I think it might work to pass the `next` path directly into the Get constructor:

```ruby
request = Net::HTTP::Get.new path
```

Or if you need to parse the URI, there might be a way to reconstruct the request.

Reply:
Updated in the latest commit, thanks!